charm/charmcraft.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: netbox
+
+type: charm
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+summary: NetBox.
+
+description: |
+    NetBox is the go-to solution for modeling and documenting network
+    infrastructure for thousands of organizations worldwide. As a successor
+    to legacy IPAM and DCIM applications, NetBox provides a cohesive,
+    extensive, and accessible data model for all things networked.
+
+extensions:
+  - django-framework
+
+requires:
+  postgresql:
+    interface: postgresql_client
+    limit: 1
+
+config:
+  options:
+    redis_hostname:
+      description: Redis hostname
+      type: string
+    redis_port:
+      description: Redis port
+      type: int
+      default: 6379
+    redis_username:
+      description: Redis username
+      type: string
+      default: ''
+    redis_password:
+      description: Redis password
+      type: string
+      default: ''
+    base_path:
```

Comment:
Let's see if we can use `self._ingress.url` instead of this.

Reply:
Updated to make it work with the self._ingress.url.I have tested manually with traefik, and it works with both path and subdomain routing. I will test it more thoroughly and add integration tests in the PR related to ingresses.